### PR TITLE
*: add the description of `alter-primary-key`

### PR DESCRIPTION
**Pull request by zimulala**
### What is changed, added or deleted? 
Add the description of `alter-primary-key` and update the description or add/drop the primary key.

### What is the related PR or file link(s)? 

- [x] Reference link(s): pingcap/tidb#13229
- [ ] This PR is to align with:
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/sql/statements/add-index.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 * 不支持 `FULLTEXT`，`HASH` 和 `SPATIAL` 索引。
 * 不支持降序索引（类似于 MySQL 5.7）。
 * 目前尚不支持同时添加多个索引。
-* 无法向表中添加 `PRIMARY KEY`。
+* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md)。
```
Comment (yikeke) [inline dev/reference/sql/statements/add-index.md]:
```suggestion
* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: dev/reference/sql/statements/create-index.md [hunk 2]
```diff
@@ -136,7 +136,7 @@ Query OK, 0 rows affected (0.31 sec)
 
 * 不支持 `FULLTEXT`，`HASH` 和 `SPATIAL` 索引。
 * 不支持降序索引 （类似于 MySQL 5.7）。
-* 无法向表中添加 `PRIMARY KEY`。
+* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md)。
```
Comment (yikeke) [inline dev/reference/sql/statements/create-index.md]:
```suggestion
* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: dev/reference/sql/statements/drop-index.md [hunk 3]
```diff
@@ -104,7 +104,7 @@ Query OK, 0 rows affected (0.30 sec)
 
 ## MySQL 兼容性
 
-* 不支持删除 `PRIMARY KEY`。
+* 默认不支持删除 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md)。
```
Comment (yikeke) [inline dev/reference/sql/statements/drop-index.md]:
```suggestion
* 默认不支持删除 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

**Comment by zimulala** [comment]
PTAL @yikeke

File: v3.0/reference/sql/statements/add-index.md [hunk 4]
```diff
@@ -101,7 +101,7 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 * 不支持 `FULLTEXT`，`HASH` 和 `SPATIAL` 索引。
 * 不支持降序索引（类似于 MySQL 5.7）。
 * 目前尚不支持同时添加多个索引。
-* 无法向表中添加 `PRIMARY KEY`。
+* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```
Comment (yikeke) [inline v3.0/reference/sql/statements/add-index.md]:
```suggestion
* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/v3.0/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: v3.0/reference/sql/statements/create-index.md [hunk 5]
```diff
@@ -136,7 +136,7 @@ Query OK, 0 rows affected (0.31 sec)
 
 * 不支持 `FULLTEXT`，`HASH` 和 `SPATIAL` 索引。
 * 不支持降序索引 （类似于 MySQL 5.7）。
-* 无法向表中添加 `PRIMARY KEY`。
+* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```
Comment (yikeke) [inline v3.0/reference/sql/statements/create-index.md]:
```suggestion
* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/v3.0/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: v3.0/reference/sql/statements/drop-index.md [hunk 6]
```diff
@@ -104,7 +104,7 @@ Query OK, 0 rows affected (0.30 sec)
 
 ## MySQL 兼容性
 
-* 不支持删除 `PRIMARY KEY`。
+* 默认不支持删除 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```
Comment (yikeke) [inline v3.0/reference/sql/statements/drop-index.md]:
```suggestion
* 默认不支持删除 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/v3.0/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: v3.1/reference/sql/statements/add-index.md [hunk 7]
```diff
@@ -101,7 +101,7 @@ EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 * 不支持 `FULLTEXT`，`HASH` 和 `SPATIAL` 索引。
 * 不支持降序索引（类似于 MySQL 5.7）。
 * 目前尚不支持同时添加多个索引。
-* 无法向表中添加 `PRIMARY KEY`。
+* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```
Comment (yikeke) [inline v3.1/reference/sql/statements/add-index.md]:
```suggestion
* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/v3.1/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: v3.1/reference/sql/statements/create-index.md [hunk 8]
```diff
@@ -136,7 +136,7 @@ Query OK, 0 rows affected (0.31 sec)
 
 * 不支持 `FULLTEXT`，`HASH` 和 `SPATIAL` 索引。
 * 不支持降序索引 （类似于 MySQL 5.7）。
-* 无法向表中添加 `PRIMARY KEY`。
+* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```
Comment (yikeke) [inline v3.1/reference/sql/statements/create-index.md]:
```suggestion
* 默认无法向表中添加 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/v3.1/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: v3.1/reference/sql/statements/drop-index.md [hunk 9]
```diff
@@ -104,7 +104,7 @@ Query OK, 0 rows affected (0.30 sec)
 
 ## MySQL 兼容性
 
-* 不支持删除 `PRIMARY KEY`。
+* 默认不支持删除 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```
Comment (yikeke) [inline v3.1/reference/sql/statements/drop-index.md]:
```suggestion
* 默认不支持删除 `PRIMARY KEY`，在开启 `alter-primary-key` 配置项后可支持此功能，详情参考：[alter-primary-key](/v3.1/reference/configuration/tidb-server/configuration-file.md#alter-primary-key)。
```

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 10]
```diff
@@ -65,6 +65,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 将旧表中的 utf8 字符集当成 utf8mb4的开关。
 + 默认值：true
 
+### `alter-primary-key`
+
++ 用于控制添加或者删除主键功能。
++ 默认值：false
++ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是 int 类型时，即使之后开启此开关也不支持对此列表删除主键。
```
Comment (tangenta) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整数类型时，即使之后开启此开关也不支持对此列表删除主键。
```

File: v3.0/reference/configuration/tidb-server/configuration-file.md [hunk 11]
```diff
@@ -66,6 +66,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 将旧表中的 utf8 字符集当成 utf8mb4的开关。
 + 默认值：true
 
+### `alter-primary-key`
+
++ 用于控制添加或者删除主键功能（在 3.0.6 及以后版本支持此配置项）。
++ 默认值：false
++ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是 int 类型时，即使之后开启此开关也不支持对此列表删除主键。
```
Comment (tangenta) [inline v3.0/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整数类型时，即使之后开启此开关也不支持对此列表删除主键。
```

File: v3.1/reference/configuration/tidb-server/configuration-file.md [hunk 12]
```diff
@@ -65,6 +65,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 将旧表中的 utf8 字符集当成 utf8mb4的开关。
 + 默认值：true
 
+### `alter-primary-key`
+
++ 用于控制添加或者删除主键功能。
++ 默认值：false
++ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是 int 类型时，即使之后开启此开关也不支持对此列表删除主键。
```
Comment (tangenta) [inline v3.1/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整数类型时，即使之后开启此开关也不支持对此列表删除主键。
```

**Pull request merged**